.github/workflows/CI.yml

```diff
@@ -103,19 +106,61 @@ jobs:
       - name: Build and push images
         uses: docker/build-push-action@v6
         with:
-          platforms: linux/amd64,linux/arm64
+          platforms: linux/${{ matrix.platform }}
           labels: ${{ steps.meta.outputs.labels }}
           tags: ${{ steps.meta.outputs.tags }}
           cache-from: type=gha
           cache-to: type=gha,mode=max
           push: true
+          provenance: false  # avoid push manifest-list
+
+  # amend the platform tag for docker images
+  docker-manifests:
+    needs: ["docker-build-push"]
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'
+    permissions:
+      packages: write  # push to ghcr.io
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get the semver
+        id: semver
+        run: |
+          VERSION="${{ github.ref_name }}"
+          SEMVER="${VERSION#v}"
+          echo "semver=${SEMVER}" >> "$GITHUB_OUTPUT"
```

Comment:
I hope I can get the tags from ${{ needs.docker-build-push.meta.tags }}, but GitHub doesn't support outputs from the specific matrix. See https://github.com/orgs/community/discussions/17245